opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala

```diff
@@ -27,29 +29,26 @@ object Tracing {
   private def currentSpan: URIO[Tracing, FiberRef[Span]] =
     ZIO.access[Tracing](_.get.currentSpan)
 
-  private def createRoot(spanName: String, spanKind: Span.Kind): URIO[Tracing, Span] =
-    ZIO.accessM[Tracing](_.get.createRoot(spanName, spanKind))
+  private def createRoot(spanName: String, spanKind: Span.Kind): URManaged[Tracing, Span] =
+    ZManaged.accessManaged[Tracing](_.get.createRoot(spanName, spanKind))
 
-  private def createChildOf(parent: Span, spanName: String, spanKind: Span.Kind): URIO[Tracing, Span] =
-    ZIO.accessM[Tracing](_.get.createChildOf(parent, spanName, spanKind))
+  private def createChildOf(parent: Span, spanName: String, spanKind: Span.Kind): URManaged[Tracing, Span] =
+    ZManaged.accessManaged[Tracing](_.get.createChildOf(parent, spanName, spanKind))
 
   private def getCurrentSpan: URIO[Tracing, Span] = currentSpan.flatMap(_.get)
 
   private def toEndTimestamp(time: Long): EndSpanOptions = EndSpanOptions.builder().setEndTimestamp(time).build()
 
-  private def endSpan(span: Span): URIO[Tracing, Unit] =
-    currentNanos.map(toEndTimestamp _ andThen span.end)
-
   private def setErrorStatus[E](span: Span, cause: Cause[E], toErrorStatus: PartialFunction[E, Status]): UIO[Unit] = {
     val errorStatus: Status = cause.failureOption.flatMap(toErrorStatus.lift).getOrElse(Status.UNKNOWN)
     UIO(span.setStatus(errorStatus.withDescription(cause.prettyPrint)))
   }
 
   /**
-   * Sets the `currentSpan` to `span` only while `effect` runs.
-   * Then ends `span` according to the result of `effect`.
+   * Sets the `currentSpan` to `span` only while `effect` runs,
+   * and error status of `span` according to any potential failure of effect.
    */
-  private def finalizeSpanUsingEffect[R, E, A](
+  private def setSpanError[R, E, A](
```

Comment:
Nit: As this does not only set an error status, I liked the previous name better.

Reply:
No problem :). Fixed with https://github.com/zio/zio-telemetry/pull/136/commits/1d528cedd20b3bd7fc688cc7d7d58ec429e7bdaf